### PR TITLE
pg 15 and 16 packer/ansible/ghactions

### PR DESCRIPTION
**Pull request by samrose**
## What kind of change does this PR introduce?

This PR is a child PR of https://github.com/supabase/postgres/pull/1266

It includes the changes to packer/ansible/gh actions to allow for a matrix build of pg versions on release.

## Significant changes

This PR moves the source of truth for data about postgres major and semver back to ansible/vars.yml, which are then read at time of release in the ami-release gh action. The _**common-vars.nix.yml file is no longer under source control**_, but instead is generated duing the ci run for ami release or testinfra runs. The version is now controlled in this allows us to avoid making too many changes to our release and test tooling for AMI)

To bump the versions of postgres it is proposed to update these values that are now in `anisble/vars.yml`

These versions cover both release and testinfra

```yml
postgres_major:
  - "15"
  - "16"

# Full version strings for each major version
postgres_release:
  postgres15: "15.8.1.003-staging-5"
  postgres16: "16.3.1.000-staging-5"
```

### CODE REVIEW

File: .github/workflows/ami-release-nix.yml [hunk 1]
```diff
@@ -3,17 +3,37 @@ name: Release AMI Nix
 on:
   push:
     branches:
-      - develop
-      - release/*
+      #- develop
+      #- release/*
+      - sam/15-16-ghactions
```
Comment (soedirgo) [inline .github/workflows/ami-release-nix.yml]: Needs reverting?

Comment (samrose) [reply .github/workflows/ami-release-nix.yml]: This is a pr to another branch where that is commented out. My intention is to merge this to that branch, once that branch pr is approved I would revert this change to trigger release

Comment (samrose) [reply .github/workflows/ami-release-nix.yml]: @soedirgo okto merge to parent pr?

Comment (soedirgo) [reply .github/workflows/ami-release-nix.yml]: Yeah parent PR looks OK, but need a +1 from infra

**Pull request merged**